creator/ingest_runs/models/validation_run.py

```diff
@@ -0,0 +1,197 @@
+import uuid
+
+import django_rq
+from django.contrib.auth import get_user_model
+from django.core.exceptions import ValidationError
+from django.db import models
+
+from creator.ingest_runs.common.model import IngestProcess
+from creator.data_reviews.models import DataReview
+from creator.files.models import Version
+
+
+User = get_user_model()
+
+
+def _get_upload_directory(validation_resultset, filename):
+    """
+    Resolves the directory where a file should be stored
+    """
+    if settings.DEFAULT_FILE_STORAGE == "django_s3_storage.storage.S3Storage":
+        prefix = f"{settings.UPLOAD_DIR}/{filename}"
+        return prefix
+    else:
+        # Validation_resultset should always have a study from its
+        # associated data_review
+        bucket = validation_resultset.study.bucket
+        directory = f"{settings.UPLOAD_DIR}/{bucket}/"
+        return os.path.join(settings.BASE_DIR, directory, filename)
+
+
+class ValidationRun(IngestProcess):
+    """
+    Request to validate file versions in a data review
+    """
+    class Meta(IngestProcess.Meta):
+        permissions = [
+            ("list_all_validationrun", "Show all validation_runs"),
+            ("cancel_validationrun", "Cancel an validation_run"),
+        ]
+
+    success = models.BooleanField(
+        null=True,
+        blank=True,
+        default=False,
+        help_text="Whether the validation run resulted in all tests passing"
+    )
+    progress = models.IntegerField(
+        null=True,
+        blank=True,
+        default=0,
+        help_text="The the percentage of the validation run that has been "
+        "completed",
+    )
+    data_review = models.ForeignKey(
+        DataReview,
+        on_delete=models.CASCADE,
+        null=True,
+        related_name="validation_runs",
+    )
+
+    @property
+    def versions(self):
+        """
+        Get file versions from the associated DataReview
+        """
+        if self.data_review:
+            return self.data_review.versions
+        else:
+            return Version.objects.none()
+
+    @property
+    def study(self):
+        """
+        Get Study from the associated data review
+        """
+        if self.data_review:
+            return self.data_review.study
```

Comment:
Same thing here about return types. Here you could return `Study.objects.none()`.

Reply:
Returning a queryset vs `None` is neat because you can treat things predictably.
Consider you want to do:
```
validation_run.study.collaborators.all()
```
This is fine if the study exists, but if you return `None`, then you break the API that the ORM provides and have to handle things differently in your code.
If you return an empty queryset, I believe you can still try to access collaborators, but operating on an empty queryset should just result in another empty queryset. If this isn't the case, then you would at least get a more formal `Study.DoesNotExist` or similar error.

Reply:
Ah yea, I meant to but forgot to return the empty queryset in a couple places